Plugins/Flow.Launcher.Plugin.Explorer/Search/WindowsIndex/IndexSearch.cs

```diff
@@ -109,27 +110,38 @@ private async static Task<List<Result>> RemoveResultsInExclusionList(List<Result
 
             var filteredResults = new List<Result>();
 
-            await Task.Run(() =>
+            try
             {
-                for (var index = 0; index < results.Count; index++)
+                await Task.Run(() =>
```

Comment:
I would say that we don't need the Task.Run because we don't need to filter result in another thread. It is sync code so using a Task.Run won't create better performance. We just need to add the token.Throw here, and this method don't need to be asynchronous.
The OperationCancelledException is supposed to be catched by PluginManager, so no need  to add another try catch here. I try to do that in WebSearch plugin because I have encoutered debugging latency (the latency only appears when debugger attach, and I haven't found a great way to resolve that), but seems that simply catching the result won't solve that issue. Therefore, no need to do that.

Reply:
> The OperationCancelledException is supposed to be catched by PluginManager, so no need to add another try catch here.

There is a try catch in IndexSearch class, how come that needs it?

Reply:
I would say no need for that, but since there already contains a try catch for ODBC exception, I add it. We can remove it to keep code cleaner.

Reply:
Sounds good, I will do that. So overall, there is no need to catch token cancellation thrown exception in plugins, because it's handled by the PluginManager, correct?

Reply:
Correct. Exception thrown is by design to avoid result return, so no need to catch them inside any plugin.